airbyte-cdk/java/airbyte-cdk/db-destinations/src/main/java/io/airbyte/cdk/integrations/destination/jdbc/typing_deduping/JdbcSqlGenerator.java

```diff
@@ -62,8 +62,10 @@ protected DataType<?> toDialectType(final AirbyteType type) {
 
   protected DataType<?> toDialectType(final AirbyteProtocolType airbyteProtocolType) {
     return switch (airbyteProtocolType) {
-      case STRING -> SQLDataType.VARCHAR;
-      case NUMBER -> SQLDataType.FLOAT;
+      // Many destinations default to a very short length (e.g. Redshift defaults to 256)
+      case STRING -> SQLDataType.VARCHAR(65535);
+      // We default to precision=39, scale=9 across destinations
+      case NUMBER -> SQLDataType.DECIMAL(38, 9);
```

Comment:
👍 and good comments

Reply:
💯

Reply:
Nit/Note - these changes do appear to be for a generic JDBC driver not redshift itself... we might want to seperate these later.